test/extensions/filters/network/redis_proxy/proxy_filter_test.cc

```diff
@@ -301,7 +301,7 @@ TEST_F(RedisProxyFilterTest, AuthWhenNotRequired) {
             error->asString() = "ERR Client sent AUTH, but no password is set";
             EXPECT_CALL(*encoder_, encode(Eq(ByRef(*error)), _));
             EXPECT_CALL(filter_callbacks_.connection_, write(_, _));
-            callbacks.onAuth("foo");
+            callbacks.onAuth("", "foo");
```

Comment:
Why is the username never non-empty in this test? Also, the password only overload (assuming it exists) should be tested as well.

Reply:
I have a new commit with a bunch of unit tests added, hopefully this flexes the code base a bit more.